### ISSUE TITLE
Custom label rotation not allowed anymore?

### ISSUE THREAD

**Issue by fanfani** [SciTools/cartopy#1566]
### Description

hello everybody, i am experiencing another issue trying to "port" some scripts that used to work just fine (i.e. as i expected) under 0.17.

first of all, let me say that reading the "What’s New in cartopy 0.18" web page i did not expect to have issues at all, as many of the changes described are presented as additions of new features.

so, i used to have vertical labels here and then and i do not get how to have this working in 0.18.

here is a comparison of 0.17 and 0.18 outputs with the same code (reported below):

0.17
![labels_017](https://user-images.githubusercontent.com/16723699/82309528-8c50cf00-99c3-11ea-9118-5d9ca3ffcb46.png)

0.18
![labels_018](https://user-images.githubusercontent.com/16723699/82309570-970b6400-99c3-11ea-8182-5886286084dd.png)

#### Code to reproduce

```
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt

import cartopy
import cartopy.io.img_tiles as cimgt
import cartopy.crs as ccrs
from cartopy.mpl.gridliner import LONGITUDE_FORMATTER, LATITUDE_FORMATTER

fig = plt.figure(figsize=(7, 5), dpi=100, frameon=False)

extent = [12, 13, 41, 42]

terrain = cimgt.GoogleTiles(url="https://server.arcgisonline.com/arcgis/rest/services/NatGeo_World_Map/MapServer/tile/{z}/{y}/{x}.png")
ax = plt.axes(projection=terrain.crs, frameon=False)
ax.set_extent(extent)
ax.add_image(terrain, 9)

gl = ax.gridlines(crs=ccrs.PlateCarree(), draw_labels=True, linewidth=0.8, color='#999999', alpha=0.5, linestyle='--')
gl.xformatter = LONGITUDE_FORMATTER
gl.yformatter = LATITUDE_FORMATTER
gl.ylabel_style = {'rotation': 90, 'size': 9, 'color': 'gray'}
gl.xlabel_style = {'size': 9, 'color': 'gray'}
gl.xpadding = 2

plt.savefig('labels.png', dpi=100)

```

In particular, setting `ylabel_style = {'rotation': someangle}` seems to trigger some new way of plotting labels in a way that is somehow misleading...

the impression you get is that cartopy it is doing something automagically, ignoring a very explicit setting (i.e. rotate the label of a specified angle)

**Comment by greglucas** [comment]
I agree, that should work. I tried a quick fix of checking if 'rotation' is in the style and skipping the magic, but then the padding is still adjusted and hides them behind the map (possibly due to a different anchor for the rotation). In addition, the angles are automatically rotated in the draw function to try and fit them in the figure, so on pan/zoom the angles can jump around even if being already set to a fixed rotation.

**Comment by QuLogic** [comment]
For the latter issue, I think you can set `gl.rotate_labels = False`. Not sure how to fix the automatic padding change though.

**Comment by stefraynaud** [comment]
In an incoming PR, rotation of labels will no longer be tested with different angles to not hide them.
That should not be difficult to test for the presence of the rotation keyword in label style.
